Marlin/stepper.cpp

```diff
@@ -601,14 +639,28 @@ void Stepper::init() {
   TCNT1 = 0;
   ENABLE_STEPPER_DRIVER_INTERRUPT();
 
-  #if ENABLED(ADVANCE)
+  #if ENABLED(ADVANCE) || ENABLED(LIN_ADVANCE)
+
+    #if ENABLED(LIN_ADVANCE)
+
+      for (int i = 0; i < EXTRUDERS; i++) {
```

Comment:
> I didn't want to completely drop the old ADVANCE feature because someone out there might find it charming 
>and want to make it work better
Yes, that's me!
>Everything I've been reading indicates that using a separate ISR for the E steppers produces nicer results, so >I'm very enthusiastic about these algorithms.

Actually you need to understand in detail how the planner works in order to spot the weaknesses, shortcomings so you can collaborate and understand how to fix it. For me, this means I require pictures to explain current issues with definitions so collaborators speak the same language. For example the velocity model: trapezoid and the relationship between velocity, acceleration and jerk. 
![image](https://cloud.githubusercontent.com/assets/8622889/15037082/ef3eef80-12da-11e6-8bdb-abd8e436b18e.png)
Researchers from universities are now writing papers about the best strategies like maximise jerk and greedy feeding algorithms. Currently I cannot optimise the feeding buffersize or change the trapezoid profile. Ideally I would like play with these and observe the quality of the print and enhance the planner maths/model. I believe there is a lot of improvement here to be made since nobody has improved the Grbl planner. See http://maxwellsci.com/print/rjaset/v5-3543-3548.pdf

At this moment of speaking @jbrazio is closing the issues with information (I must say, in most cases it is justified). That means for me, some research or learning is lost. Conclusions and learnings need to be captured in a wiki so that can be the spec for designing and building the overhaul for a complex item (not a bug fix or simple additional feature). Anyway that is just my opinion based on my way of learning and designing.